packages/block-library/src/image/interactivity.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * Internal dependencies
+ */
+import { store } from '../utils/interactivity';
+
+const raf = window.requestAnimationFrame;
+// Until useSignalEffects is fixed: https://github.com/preactjs/signals/issues/228
+const tick = () => new Promise( ( r ) => raf( () => raf( r ) ) );
+
+const focusableSelectors = [
+	'a[href]',
+	'area[href]',
+	'input:not([disabled]):not([type="hidden"]):not([aria-hidden])',
+	'select:not([disabled]):not([aria-hidden])',
+	'textarea:not([disabled]):not([aria-hidden])',
+	'button:not([disabled]):not([aria-hidden])',
+	'iframe',
+	'object',
+	'embed',
+	'[contenteditable]',
+	'[tabindex]:not([tabindex^="-"])',
+];
+
+store( {
+	actions: {
+		core: {
+			image: {
+				showLightbox: ( { context } ) => {
+					context.core.image.initialized = true;
+					context.core.image.lightboxEnabled = true;
+					context.core.image.lastFocusedElement =
+						window.document.activeElement;
+					context.core.image.scrollPosition = window.scrollY;
+
+					context.core.image.documentElement =
+						document.documentElement;
+					context.core.image.adminElement =
+						document.querySelector( '#wpadminbar' );
+					context.core.image.siteBlocksElement =
+						document.querySelector( '.wp-site-blocks' );
+
+					context.core.image.documentElement.classList.add(
+						'has-lightbox-open'
+					);
+
+					if ( context.core.image.adminElement ) {
+						context.core.image.adminElement.setAttribute(
+							'inert',
+							true
+						);
+					}
+
+					if ( context.core.image.siteBlocksElement ) {
+						context.core.image.siteBlocksElement.setAttribute(
+							'inert',
+							true
+						);
+					}
```

Comment:
Is it needed to differentiate between admin and not admin? I don't like too much sending code only relevant to the admin for all the page visits.

Sorry if it doesn't make sense but, can't we just add the `inert` property to the body in both cases?

Reply:
> Is it needed to differentiate between admin and not admin? I don't like too much sending code only relevant to the admin for all the page visits.

Nope, it's done this way because I'm trying to target the two elements separately.

> Sorry if it doesn't make sense but, can't we just add the inert property to the body in both cases?

We can't, because the lightbox is a child of the body and will also become inert in that case.

I was hoping to get feedback on what would be the best way to mark the appropriate DOM elements as inert, or if it's even necessary if we've already marked the dialog as being open, have trapped focus, and have added a scrim to show the background elements as inert. Also, targeting these two HTML elements doesn't guarantee that other interactive elements won't be on the page. For example, at the moment it's not capturing the 'Skip to Content' button or other elements that may be added dynamically to the DOM.

At the same time, creating logic to loop through children of the body to mark them as inert seemed like overkill.

Perhaps @andrewhayward can offer guidance?

Reply:
Okay, now I understand 🙂 Thanks for the comment!

I'm far from an expert in accessibility but, as you say, marking the dialog, trapping the focus, and adding a background covering the whole screen, seem enough for me. Actually, the current navigation block is not using inert and it follows a similar approach to this one. Anyway, I'd love to understand if/why inert is needed.

> At the same time, creating logic to loop through children of the body to mark them as inert seemed like overkill.

If we finally have to include `inert` to all the elements that are not the lightbox, it seems safer to include it to all the children of the body that are not the current lightbox, right? Maybe some plugins or themes are adding other elements that we are not considering. Something like this: 

```js
for (const element of document.querySelector("body").children) {
  !(element === context.core.image.currentLightbox) &&
    element.setAttribute("inert", "");
}
```

Reply:
Given that the lightbox already follows a similar pattern to the navigation block, I've gone ahead and removed the `inert` attribute logic for now pending additional discussions. Since this PR is coming out with the lightbox disabled by default and as part of an experiment, the plan is to continue testing accessibility more widely and continue making improvements, so we can always add this back in.